src/OSes/ubuntu.py

```diff
@@ -13,14 +14,58 @@
     DNSMASQ_LOCAL_CONF = DNSMASQ_LOCAL_CONF.replace('dnsmasq.d', 'conf.d')
 
 
-def setup(tld=config.TOP_LEVEL_DOMAIN):
-    return None
+def setup(tld=config.TOP_LEVEL_DOMAIN) -> int:
+    return 0
+
+
+def install(tld=config.TOP_LEVEL_DOMAIN) -> int:
+    try:
+        resolvconf = '/etc/resolv.conf'
+        resolvconf_head = '/etc/resolvconf/resolv.conf.d/head'
+
+        if not os.path.exists(resolvconf):
+            content = '# Generated by docker-dns\n'
+            open(resolvconf, 'w').write(content)
 
+        dns = docker.NETWORK_GATEWAY
+        resolvconf_new_lines = [
+            f'nameserver {dns} #@docker-dns\n',
+            'nameserver 1.1.1.1 #cloudflare #@docker-dns\n',
+            'nameserver 8.8.8.8 #google #@docker-dns\n',
+            'options timeout:1 #@docker-dns\n'
+        ]
+        if int(config.OS_VERSION.split('.')[0]) >= 18:
+            resolvconf_head_lines = open(resolvconf_head, 'r').readlines()
+            resolvconf_new_lines_filtered = [
+                line for line in resolvconf_new_lines
+                if line not in resolvconf_head_lines
+            ]
```

Comment:
```suggestion
            resolvconf_new_lines_filtered = [line for line in resolvconf_new_lines if line not in resolvconf_head_lines]
```
Acho que usar um builder de array em linhas quebradas pode deixar o código um pouco confuso

Reply:
Acho ela quebrada masi fácil de ler mas isso acaba sendo sempre um pouco subjetivo